### PR TITLE
Fix rule filtering in the document walker

### PR DESCRIPTION
**Pull request by mclem**
### 🤔 What's changed?

Rules are now filtered when their all their children (background and scenarios) are themselves filtered.

### ⚡️ What's your motivation? 

Fix for https://github.com/cucumber/react-components/issues/136

### 🏷️ What kind of change is this?

- :bug: Bug fix (non-breaking change which fixes a defect)

### ♻️ Anything particular you want feedback on?

I have no experience in TypeScript so I'm not sure about the fix. It seems that the bug was about a difference in handling of `undefined` vs `null`.
Please feel free to modify anything as required.

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: javascript/src/GherkinDocumentWalker.ts [hunk 1]
```diff
@@ -201,8 +201,8 @@ export default class GherkinDocumentWalker {
 
     this.handlers.handleRule(rule)
 
-    const backgroundKept = children.find((child) => child !== null && child.background !== null)
-    const scenariosKept = children.filter((child) => child !== null && child.scenario !== null)
+    const backgroundKept = children.find((child) => child !== null && child.background)
```
Comment (xeger) [inline javascript/src/GherkinDocumentWalker.ts:204]: I know you're not an expert, but: any thoughts on whether we should relax the `child !== null` comparison in a similar way?

If `child` is always an object, then it's probably a better idea to check for truthiness (is not null or undefined) than to explicitly check for null, even if `child` is not ever expected to take an `undefined` value. (After all, `child.background` and `child.scenario` were expected to never take `undefined`, yet here we now are...)

**Review by xeger — CHANGES_REQUESTED** [review]
Please add a brief description of how the behavior has changed to the `Unreleased` section of `CHANGELOG`.

Linking to the upstream cucumber issue when you describe the fix would be very helpful, but mainly, we want to make sure people will have an incentive to upgrade to the next release by knowing which specific bugs have been fixed, without researching the closed pull requests, etc.

**Comment by luke-hill** [comment]
@mpkorstanje / @Dbuggerx can this now be closed?

**Comment by mpkorstanje** [comment]
Superseded by https://github.com/cucumber/gherkin-utils/pull/48

**Pull request closed without merging**